### PR TITLE
Use npm shrinkwrap and travis caching

### LINKED ISSUE #7820: re-enable npm travis caching

### PR DESCRIPTION
**Pull request by diracdeltas**
fix #7820 

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).

Test Plan:
1. automated tests should not fail with a lint error

### CODE REVIEW

**Comment by bsclifton** [comment]
Docs for reference (I've never used shrinkwrap- need to look more into it) - https://docs.npmjs.com/cli/shrinkwrap

This would make the updating processing interesting.  I'm also curious if it'll work OK across platforms

**Comment by cezaraugusto** [comment]
fyi I opened a PR to update standard@9 that could help as well: #7817 so repo/travis can share the same version

**Comment by cezaraugusto** [comment]
sorry just saw [Yan comment](https://github.com/brave/browser-laptop/pull/7817#issuecomment-288216178) on ref PR. I'm curious why Travis isn't matching repo's deps version.

**Comment by bsclifton** [comment]
@diracdeltas would you be able to run `npm shrinkwrap --dev` again? (and commit/squash)

I think this PR solves the problem in a much more standard way than https://github.com/brave/browser-laptop/issues/8283.  Sorry I had been hesitant to accept it- this would be a great one 😄

**Comment by diracdeltas** [comment]
updated shrinkwrap

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -22,6 +22,7 @@ Brave welcomes contributions via [pull requests](https://github.com/brave/browse
     Fix #206
 ````
 
+* If you update the npm dependencies, run `npm shrinkwrap --dev` before committing changes.
```
Comment (luixxiul) [inline .github/CONTRIBUTING.md]: Question: is it ok to run that everytime when I commit something to keep the dependencies up to date?

Comment (bsclifton) [inline .github/CONTRIBUTING.md]: @luixxiul we'll want to pick and choose the times we update dependencies. Unless there is a good reason (need new version to get a feature, we're too out of date, etc) , we should stick with the version there.  This is important to stabilize the codebase.  

Any updates we do to libraries we use may require a small regression test to ensure the upgrade didn't break any functionality or introduce issues

Comment (luixxiul) [inline .github/CONTRIBUTING.md]: @bsclifton thanks!

**Pull request merged**